packages/components/Molecules/Search/__tests__/RplSearchForm.test.js

```diff
@@ -0,0 +1,78 @@
+import { shallowMount } from '@vue/test-utils'
+import SearchForm from '../SearchForm'
+
+describe('SearchForm', () => {
+  it('emits search event with a value when submitting the form', () => {
+    const wrapper = shallowMount(SearchForm)
+    wrapper.setProps({ allowBlank: false })
+    wrapper.setData({ searchInput: 'test keyword' })
+
+    wrapper.vm.submitSearch()
+    expect(wrapper.emitted().search[0]).toEqual(['test keyword'])
+  })
+
+  it('doesn\'t emit search event when there is no search input', () => {
+    const wrapper = shallowMount(SearchForm)
+    wrapper.setData({ searchInput: '' })
+
+    wrapper.vm.submitSearch()
+    expect(wrapper.emitted().search).toBeUndefined()
+  })
+
+  it('clears form input when triggered', () => {
+    const wrapper = shallowMount(SearchForm)
+    wrapper.setData({ searchInput: 'something' })
+
+    wrapper.vm.clearSearch()
+    expect(wrapper.vm.searchInput).toBe('')
+  })
+
+  it('tests searchFormClass to return empty array when there is no theme assigned', () => {
+    const wrapper = shallowMount(SearchForm)
+
+    wrapper.setProps({ theme: '' })
+    expect(wrapper.vm.searchFormClass).toEqual([])
+  })
+
+  it('adds classes to form when a valid theme is assigned', () => {
+    const wrapper = shallowMount(SearchForm)
+
+    wrapper.setProps({ theme: 'solid' })
+    expect(wrapper.vm.searchFormClass).toEqual(['rpl-search-form--solid'])
+  })
+
+  it('returns total number of form filters correctly', () => {
+    const wrapper = shallowMount(SearchForm)
+
+    wrapper.setProps({ filterForm: {} })
+    expect(wrapper.vm.filterCount).toEqual(0)
+
+    wrapper.setProps({
+      filterForm: {
+        model: {
+          testOne: ['one', 'two', 'three'],
+          testTwo: ['one', 'two', 'three']
+        }
+      }
+    })
+    expect(wrapper.vm.filterCount).toEqual(2)
+
+    wrapper.setProps({
+      filterForm: {
+        model: {
+          testOne: 'test',
+          testTwo: ['one', 'two', 'three'],
+          testThree: ''
+        }
+      }
+    })
+    expect(wrapper.vm.filterCount).toEqual(2)
+  })
+
+  it('tests watcher for prefillSearchTerm to assign value to searchInput correctly', () => {
+    const wrapper = shallowMount(SearchForm)
+
+    wrapper.setProps({ prefillSearchTerm: 'something' })
+    expect(wrapper.vm.searchInput).toEqual('something')
+  })
+})
```

Comment:
I was a bit fiddly and added this. Not directly related to ticket but ran into this when I was debugging search.

Reply:
> I was a bit fiddly and added this. Not directly related to ticket but ran into this when I was debugging search.

That looks nice to me. 👍  No need to cover all logics in this ticket. That will require a lot effort and background knowledge. We can add tests for any search bug fixes later.